src/debugger/Debugger.js

```diff
@@ -197,7 +197,15 @@ export class DebugServer {
         column: column,
       };
       frameInfos.push(frameInfo);
-      loc = frame.loc;
+      if (frame.loc) {
```

Comment:
This source loc parsing code is duplicated in at least two places. Consider putting it inside a helper function.